source/apiVolontaria/apiVolontaria/views.py

```diff
@@ -127,7 +127,11 @@ def get_queryset(self):
         return User.objects.filter()
 
     def get(self, request, *args, **kwargs):
-        if self.request.user.has_perm('apiVolontaria.get_user'):
+        if 'profile' in self.kwargs.keys():
+            self.kwargs['pk'] = self.request.user.id
+            return self.retrieve(request, *args, **kwargs)
```

Comment:
I don't completely understand this part. Why would you set a kwargs `pk` as the calling user's id ?

Reply:
As you have seen in the `urls.py` i've add a new `profile` kwargs to force a new behaviour if you come from the `/profile` url.

If you come from this special url, you want to have your own profile and not a profile from an other user (you don't specify a pk). So i force the required `pk` kwargs to the pk of the current authenticated user.